### PR TITLE
feat: Add ability to set route to multiple host

### PR DESCRIPTION
**Pull request by mrjk990**
Example
```php
$routes->group('api', ['hostname' =>  ['api.website.com', 'api.website.net]], function ($routes) {
    $routes->get('/', 'Api::royplayer');
});
```

**Description**
in this way i can set route for multiple host instead of creating multiple route for each hostname

**Checklist:**
- [x] Securely signed commits
- [ ] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [ ] User guide updated
- [ ] Conforms to style guide

### CODE REVIEW

**Comment by kenjis** [comment]
Thank you for sending this PR!

We expect the following in all Pull Requests (PRs).
- PRs must be sent to the [appropriate branch](https://github.com/codeigniter4/CodeIgniter4/blob/develop/contributing/pull_request.md#branching)
- All git commits must be [GPG-signed](https://github.com/codeigniter4/CodeIgniter4/blob/develop/contributing/pull_request.md#signing)
- Must follow our [style guide](https://github.com/codeigniter4/CodeIgniter4/blob/develop/contributing/pull_request.md#php-style)
- Be [commented](https://github.com/codeigniter4/CodeIgniter4/blob/develop/contributing/pull_request.md#comments) in the PHP source file
- Be documented in the [user guide](https://github.com/codeigniter4/CodeIgniter4/blob/develop/contributing/pull_request.md#user-guide)
- Be [unit tested](https://github.com/codeigniter4/CodeIgniter4/blob/develop/contributing/pull_request.md#unit-testing)
- Pass all checks in GitHub Actions

> [!IMPORTANT]
> We expect all code changes or bug-fixes to be accompanied by one or more tests added to our test suite to prove the code works.

If pull requests do not comply with the above, they will likely be closed. Since we are a team of volunteers, we don't have any more time to work on the framework than you do. Please make it as painless for your contributions to be included as possible.

See https://github.com/codeigniter4/CodeIgniter4/blob/develop/contributing/pull_request.md

File: system/Router/RouteCollection.php [hunk 1]
```diff
@@ -1565,7 +1565,20 @@ private function checkHostname($hostname): bool
...
+        // Hostname has multiple hostname
+        if(is_array($hostname)) 
+        {
+            $hasMatch = false;
+            foreach ($hostname as $host) {
+                if (strtolower($this->httpHost) === strtolower($host)) {
+                    $hasMatch = true;
+                }
+            }
+            return $hasMatch;
```
Comment (ddevsr) [inline system/Router/RouteCollection.php:1577]:
```suggestion
            $hostnameLower = array_map('strtolower', $hostname);

            return in_array(strtolower($this->httpHost), $hostnameLower, true);
```

No need looping, by the way needed unit testing for this

**Review by kenjis — CHANGES_REQUESTED** [review]
We do not accept new features in `develop` branch.
Please change the base branch to `4.6`.
See https://github.com/codeigniter4/CodeIgniter4/blob/develop/contributing/workflow.md#if-you-sent-to-the-wrong-branch

**Comment by kenjis** [comment]
> [!IMPORTANT]
> We expect all code changes or bug-fixes to be accompanied by one or more tests
> added to our test suite to prove the code works.

This PR does not have the necessary test code. If you want this PR to be reviewed, please add the test code.
See https://github.com/codeigniter4/CodeIgniter4/blob/develop/contributing/pull_request.md#unit-testing

If the test is not added, this PR will be closed.

**Pull request closed without merging**